crates/dbs-virtio-devices/src/mmio/mmio_state.rs

```diff
@@ -346,20 +346,32 @@ where
         if self.device_activated {
             warn!("reset device while it's still in active state");
         } else {
+            let mut reset_suc_flag = true;
             // . Keep interrupt_evt and queue_evts as is. There may be pending
             //   notifications in those eventfds, but nothing will happen other
             //   than supurious wakeups.
             // . Do not reset config_generation and keep it monotonically increasing
+
             for queue in self.queues.iter_mut() {
-                queue.queue = Q::new(queue.queue.max_size());
+                let new_queue = Q::new(queue.queue.max_size());
+                if let Err(e) = new_queue {
+                    reset_suc_flag = false;
+                    warn!("reset device failed because new virtio-queue could not be created due to {:?}", e);
```

Comment:
If one of the Queues fails to initialize, the device does not work.

I think is it better to panic directly? @jiangliu

Reply:
Or modify the `reset` function to return `Result`, and the caller decides the next action.

Reply:
`Result` is added for `reset`.
And how the caller handle `reset` 's `Result` is also updated.